### PR TITLE
[timeseries] refactor Chronos model, force unit tests to use local cache

### PR DESCRIPTION
**Pull request by canerturkmen**
*Issue #, if available:*

*Description of changes:*

This PR takes care of several housekeeping items for timeseries.
- We force Chronos unit tests to use a local cache in most test invocations, cutting down the calls to HuggingFace during CI by a factor of ~20x. This is required to prevent spurious download numbers showing up on HF due to AG CI.
- Rename the pipeline filename to prevent `chronos.chronos` namespace.
- Makes default configurations used by amazon models explicit
- Applies latest changes from the amazon-science repo. Specifically https://github.com/amazon-science/chronos-forecasting/pull/24/files and https://github.com/amazon-science/chronos-forecasting/pull/25/files

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: timeseries/src/autogluon/timeseries/models/chronos/model.py [hunk 1]
```diff
@@ -200,8 +217,32 @@ def _is_gpu_available(self) -> bool:
...
+    def ag_default_config(self) -> Dict[str, Any]:
+        """The default configuration of the model used by AutoGluon if the model is one of those
+        defined in MODEL_CONFIGS. For now, these are ``amazon/chronos-t5-*`` family of models.
+        """
+        return MODEL_CONFIGS.get(self.model_path, {})
+
+    @property
+    def min_num_gpus(self) -> int:
+        """Minimum number of GPUs required for the model. For models not defined in AutoGluon,
+        this value defaults to 8.
```
Comment (abdulfatir) [inline timeseries/src/autogluon/timeseries/models/chronos/model.py:229]: defaults to 0?

File: timeseries/tests/conftest.py [hunk 2]
```diff
@@ -31,3 +31,30 @@ def pytest_collection_modifyitems(config, items):
...
+    try:
+        from autogluon.timeseries.models.chronos.pipeline import ChronosPipeline
+
+        # download and cache model from hf hub
+        _ = ChronosPipeline.from_pretrained(model_hub_id, cache_dir=str(cache_path))
+
+        # get model snapshot path
+        snapshots_path = cache_path / "models--amazon--chronos-t5-tiny" / "snapshots"
+        assert snapshots_path.exists()
+        snapshot_dir = next(snapshots_path.iterdir())
```
Comment (abdulfatir) [inline timeseries/tests/conftest.py:55]: Can there be multiple directories in `snapshots`?

Comment (abdulfatir) [reply timeseries/tests/conftest.py:55]: You can actually:

```
(base) ➜  snapshots ll /home/ubuntu/.cache/huggingface/hub/models--amazon--chronos-t5-base/snapshots
total 8.0K
drwxrwxr-x 2 ubuntu ubuntu 4.0K Mar 27 19:18 45d3b2421f623c7239897e753976bc883cb522d0
drwxrwxr-x 2 ubuntu ubuntu 4.0K Mar 20 17:03 c735c3bee35757ba75c4f11f0153b9838ad338de
```
Maybe read the commit hash for the main ref and use that?
```
(base) ➜  refs cat /home/ubuntu/.cache/huggingface/hub/models--amazon--chronos-t5-base/refs/main 
c735c3bee35757ba75c4f11f0153b9838ad338de%
```

Comment (abdulfatir) [reply timeseries/tests/conftest.py:55]: But maybe in the scope of CI you would only have one `main` ref and hash, so the current code should be okay.

File: timeseries/src/autogluon/timeseries/models/chronos/model.py (hunk 1 above)
Comment (canerturkmen) [reply timeseries/src/autogluon/timeseries/models/chronos/model.py]: thanks for catching this!!

**Comment by yinweisu** [comment]
| Previous CI Run | Current CI Run |
| :---: | :---: |

File: timeseries/tests/conftest.py [hunk 3]
```diff
@@ -31,3 +31,30 @@ def pytest_collection_modifyitems(config, items):
 def temp_model_path(tmp_path_factory):
     fn = tmp_path_factory.mktemp(str(uuid4())[:6])
     return str(fn)
+
+
+@pytest.fixture(scope="session")
+def hf_model_path(tmp_path_factory):
```
Comment (shchur) [inline timeseries/tests/conftest.py:37]: `tmp_path_factory` is doing some heavy lifting recently 😄

File: timeseries/src/autogluon/timeseries/models/chronos/pipeline.py [hunk 4]
```diff
@@ -288,6 +313,48 @@ def __init__(self, tokenizer, model):
...
+            provided, and the extra 1 is for EOS.
+        """
+        context = self._prepare_and_validate_context(context=context)
+        token_ids, attention_mask, tokenizer_state = self.tokenizer.input_transform(context)
+        embeddings = self.model.encode(
+            input_ids=token_ids.to(self.model.device),
+            attention_mask=attention_mask.to(self.model.device),
+        ).cpu()
+        return embeddings, tokenizer_state
+
```
Comment (shchur) [inline timeseries/src/autogluon/timeseries/models/chronos/pipeline.py:357]: Do I understand correctly that this is the most recent state of `amazon-science/chronos-forecasting` without any major edits from our side?

**Review by shchur — APPROVED** [review]
LGTM! Just one question

File: timeseries/src/autogluon/timeseries/models/chronos/pipeline.py (hunk 4 above)
Comment (canerturkmen) [reply timeseries/src/autogluon/timeseries/models/chronos/pipeline.py:357]: yes, exactly

**Pull request merged**